File: zombie-survivor/typescript/src/Level.ts
export enum Level {
  Blue = 'Blue',
  Yellow = 'Yellow',
  Orange = 'Orange',
  Red = 'Red',
}

export function levelFor(xp: number): Level {
  if (xp >= 43) return Level.Red;
  if (xp >= 19) return Level.Orange;
  if (xp >= 7) return Level.Yellow;
  return Level.Blue;
}

const LevelOrder: Record<Level, number> = {
  [Level.Blue]: 0,
  [Level.Yellow]: 1,
  [Level.Orange]: 2,
  [Level.Red]: 3,
};

export function maxLevel(a: Level, b: Level): Level {
  return LevelOrder[a] >= LevelOrder[b] ? a : b;
}

File: zombie-survivor/python/src/zombie_survivor/level.py
from enum import Enum


class Level(Enum):
    BLUE = "Blue"
    YELLOW = "Yellow"
    ORANGE = "Orange"
    RED = "Red"


_LEVEL_ORDER = {Level.BLUE: 0, Level.YELLOW: 1, Level.ORANGE: 2, Level.RED: 3}


def level_for(xp: int) -> Level:
    if xp >= 43:
        return Level.RED
    if xp >= 19:
        return Level.ORANGE
    if xp >= 7:
        return Level.YELLOW
    return Level.BLUE


def max_level(a: Level, b: Level) -> Level:
    return a if _LEVEL_ORDER[a] >= _LEVEL_ORDER[b] else b
